### PR TITLE
Add IPv4 validation

### PR DESCRIPTION
**Pull request by JosipBeDa**

### CODE REVIEW

File: src/main/java/net/barrage/tegridy/validation/validator/Ip4Validator.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package net.barrage.tegridy.validation.validator;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import net.barrage.tegridy.validation.annotation.Ip4;
+
+/** Validates whether the given string field is a valid IPv4 address. */
+public class Ip4Validator implements ConstraintValidator<Ip4, Object> {
+  static final String IP4_REGEX = "^((25[0-5]|(2[0-4]|1\\d|[1-9]|)\\d)\\.?\\b){4}$";
```
Comment (blipovac) [inline src/main/java/net/barrage/tegridy/validation/validator/Ip4Validator.java:9]: Do we want to allows notation with subnets mask, or maybe create separate validation for ips with subnet mask?

File: src/test/java/validation/IpTests.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    assertEquals(1, violations2.size());
+
+    var qux = new Foo("0.0.0.0.0.0");
+    var violations3 = validator.validate(bar);
+    assertEquals(1, violations3.size());
+
+    var quz = new Foo("192.168.0.69/24");
+    var violations4 = validator.validate(quz);
+    assertEquals(1, violations4.size());
+  }
```
Comment (blipovac) [inline src/test/java/validation/IpTests.java:57]: you know I like to be nit-picky :P, but should we add message assertions as well?

**Review by blipovac — COMMENTED** [review]
Great work, thanks for adding this <3

File: src/main/java/net/barrage/tegridy/validation/validator/Ip4Validator.java (hunk 1 above)
Comment (JosipBeDa) [reply src/main/java/net/barrage/tegridy/validation/validator/Ip4Validator.java:9]: Definitely separate

**Pull request merged**